core/controllers/firebase.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Firebase proxy controller."""
+
+from __future__ import annotations
+
+import json
+
+from core import feconf
+from core.controllers import acl_decorators
+from core.controllers import base
+
+import requests
+from typing import Any, Dict
+
+
+TIMEOUT_SECS = 60
+"""Timeout in seconds for firebase requests."""
+
+FIREBASE_DOMAINS = {
+    'oppiaserver-backup-migration.appspot.com':
+        'https://oppiaserver-backup-migration.firebaseapp.com',
+    'www.oppiatestserver.org': 'https://oppiatestserver.firebaseapp.com',
+    'www.oppia.org': 'https://oppiaserver.firebaseapp.com'
+}
+"""A mapping of oppia domain to firebase domains used for authentication."""
+
+
+class FirebaseProxyPage(
+    base.BaseHandler[Any, Any]
```

Comment:
Why `Any` is required here?

I think we have a defined set of args.

Reply:
done